system/jlib/jstring.cpp

```diff
@@ -960,13 +961,16 @@ StringBuffer &replaceString(StringBuffer & result, size_t lenSource, const char
             else
                 offset++;
         }
-        // Append the remaining characters
-        result.append(lenSource - lastCopied, source + lastCopied);
+
+        if (lastCopied || !avoidCopyIfUnmatched)
+            result.append(lenSource - lastCopied, source + lastCopied); // Append the remaining characters
+
+        return lastCopied;
```

Comment:
trivial: Change to "return lastCopied != 0;" to prevent warnings about implicit conversion of integer to boolean.

Reply:
Fixed.